src/module.c

```diff
@@ -2079,6 +2079,12 @@ int RM_BlockedClientMeasureTimeEnd(RedisModuleBlockedClient *bc) {
  * the -LOADING error)
  */
 void RM_Yield(RedisModuleCtx *ctx, int flags, const char *busy_reply) {
+    static int yield_nesting = 0;
+    /* Avoid nested calls to RM_Yield */
+    if (yield_nesting)
+        return;
```

Comment:
I still prefer an assert here, the nested yield call is a wrong usage and it's a new API, asset doesn't crash anyone and can help module developer to find problem when testing.

Reply:
ok...